spec/requests/api/replacement_document_validation_request_spec.rb

```diff
@@ -6,72 +6,123 @@
-              "old_document" => {
-                "name" => "proposed-floorplan.png",
-                "invalid_document_reason" => "Document is invalid",
-                "url" => json["data"][0]["old_document"]["url"]
-              }
-            },
-            {
-              "id" => replacement_document_validation_request2.id,
-              "state" => "closed",
-              "response_due" => replacement_document_validation_request2.response_due.to_s(:db),
-              "days_until_response_due" => 15,
-              "cancel_reason" => nil,
-              "cancelled_at" => nil,
-              "old_document" => {
-                "name" => "proposed-floorplan.png",
-                "invalid_document_reason" => "Document is invalid",
-                "url" => json["data"][1]["old_document"]["url"]
-              },
-              "new_document" => {
-                "name" => "proposed-floorplan.png",
-                "url" => json["data"][1]["new_document"]["url"]
-              }
+      end
+
+      it "returns the request data" do
+        travel_to(DateTime.new(2022, 1, 2)) do
+          get(
+            path,
+            params: { change_access_id: planning_application.change_access_id },
+            headers: { "CONTENT-TYPE": "application/json", Authorization: token }
+          )
+        end
+
+        expect(json["data"]).to contain_exactly(
+          {
+            id: replacement_document_validation_request.id,
+            state: "open",
+            response_due: "2022-01-25",
+            days_until_response_due: 15,
+            cancel_reason: nil,
+            cancelled_at: nil,
+            old_document: {
+              name: "proposed-floorplan.png",
+              invalid_document_reason: "Document is invalid",
+              url: "http://www.example.com/test_image"
+            }
+          }.deep_stringify_keys,
+          {
+            id: replacement_document_validation_request2.id,
+            state: "closed",
+            response_due: "2022-01-25",
+            days_until_response_due: 15,
+            cancel_reason: nil,
+            cancelled_at: nil,
+            old_document: {
+              name: "proposed-floorplan.png",
+              invalid_document_reason: "Document is invalid",
+              url: "http://www.example.com/test_image"
             },
-            {
-              "id" => replacement_document_validation_request3.id,
-              "state" => "cancelled",
-              "response_due" => replacement_document_validation_request3.response_due.to_s(:db),
-              "days_until_response_due" => 15,
-              "cancel_reason" => "Made by mistake!",
-              "cancelled_at" => json_time_format(replacement_document_validation_request3.cancelled_at),
-              "old_document" => {
-                "name" => "proposed-floorplan.png",
-                "invalid_document_reason" => nil,
-                "url" => json["data"][2]["old_document"]["url"]
```

Comment:
I've refactored these tests a bit because this is kind of saying `expect(json).to eq({ foo: json[:foo] }`, which will pass whatever the value of `foo` is!